test/test_AIS.cpp

```diff
@@ -254,3 +264,168 @@ TEST_F(AISTest, it_converts_marnav_message05_into_a_VoyageInformation)
     ASSERT_EQ(7890, info.imo);
     ASSERT_EQ("DEST", info.destination);
 }
+
+TEST_F(AISTest, it_selects_yaw_as_vessel_heading_source_if_available)
+{
+    ais::message_01 msg;
+    msg.set_sog(0);
+    msg.set_hdg(90);
+    auto position = AIS::getPosition(msg);
+
+    auto [ori, status] = AIS::selectVesselHeadingSource(position.yaw,
+        position.course_over_ground,
+        position.speed_over_ground);
+
+    ASSERT_NEAR(ori.z(), -0.7071068, 1e-3);
+    ASSERT_NEAR(ori.w(), 0.7071068, 1e-3);
```

Comment:
hard to tell if ir is right or not, I would have to know Quaternions for that, the following is more clear:

```suggestion
    ASSERT(ori.isApprox(Eigen::Quaterniond(Eigen::AngleAxisd(-M_PI / 2, Eigen::Vecotr3d::UnitZ()));
```